app/scripts/metamask-controller.js

```diff
@@ -4543,6 +4554,10 @@ export default class MetamaskController extends EventEmitter {
         config.type !== networkConfigurationId,
     );
 
+    const scope = `eip155:${parseInt(chainId, 16)}`;
+    this.multichainSubscriptionManager.unsubscribe(scope);
+    this.multichainMiddlewareManager.removeMiddleware(scope);
```

Comment:
this gets rid of middleware and subscription when network is removed but im sure there are other places we want to do it, like when re-requesting caip-25 permissions

Reply:
we probably need to add a stateChange listener to PermissionController too